docs/shared/recipes/module-federation/federate-a-module.md

```diff
@@ -0,0 +1,198 @@
+```shell
+npm i -D @nx/react
+```
+
+Add Host with Remote
+
+```shell
+nx generate @nx/react:host --name=home --remotes=greeting --projectNameAndRootFormat=as-provided
+```
+
+{% /tab %}
+{%tab label="Angular"%}
+
+Add Angular
+
+```shell
+npm i -D @nx/angular
+```
+
+Add Host with Remote
+
+```shell
+nx generate @nx/angular:host --name=home --remotes=greeting --projectNameAndRootFormat=as-provided
+```
+
+{% /tab %}
+{% /tabs %}
+
+## Step 2: Create the module
+
+In this example, we will create a module that exports a hello function.
+Since we are using Nx, we will create a library for this module.
+
+**Create a library**
+
+```shell
+nx generate @nx/js:library --name=hello --unitTestRunner=jest --projectNameAndRootFormat=as-provided
+```
+
+Update the `hello.ts` file with the following code:
+
+```typescript {% fileName="hello/src/lib/hello.ts" %}
+export default function hello(): string {
+  return 'Hello from Nx';
+}
+```
+
+Update `hello/` barrel file `index.ts` with the following code:
+
+```typescript {% fileName="hello/src/index.ts" %}
+export { default } from './lib/hello';
+```
+
+## Step 3: Federate the module
+
+Now that we have created the module, we need to configure it to be federated.
+
+{% tabs %}
+{%tab label="React"%}
+
+```shell
+nx generate @nx/react:federate-module Hello --remote=greeting --path=hello/src/index.ts --projectNameAndRootFormat=as-provided
+```
+
+{% /tab %}
+{%tab label="Angular"%}
+
+```shell
+nx generate @nx/angular:federate-module Hello --remote=greeting --path=hello/src/index.ts --projectNameAndRootFormat=as-provided
+```
+
+{% /tab %}
+{% /tabs %}
+
+{% callout type="note" title="Remote does not exist" %}
+If the remote provided does not exist (in this instance _greeting_), Nx will create it for you.
+{% /callout %}
+
+This command will:
+
+- Add a module entry to the `greeting` remote module federation config file.
+
+{% tabs %}
+{%tab label="typescript"%}
+
+```typescript {% fileName="greeting/module-federation.config.ts" %}
+import { ModuleFederationConfig } from '@nx/webpack';
+
+const config: ModuleFederationConfig = {
+  name: 'greeting',
+  exposes: {
+    './Module': './src/remote-entry.ts',
+    './Hello': 'hello/src/index.ts', // <-- this line was added,
+  },
+};
+export default config;
+```
+
+{% /tab %}
+{%tab label="javascript"%}
+
+```javascript {% fileName="greeting/module-federation.config.js" %}
+module.exports = {
+  name: 'greeting',
+  exposes: {
+    './Module': './src/remote-entry.ts',
+    './Hello': 'hello/src/index.ts', // <-- this line was added
+  },
+};
+```
+
+{% /tab %}
+{% /tabs %}
+
+- Add a typescript path mapping to the `greeting/Hello` module in `tsconfig.json` file.
+
+```json {% fileName="/tsconfig.base.json" %}
+{
+  "paths": {
+    "greeting/Module": ["greeting/src/remote-entry.ts"],
+    "greeting/Hello": ["hello/src/index.ts"] // <-- this line was added
+  }
+}
+```
+
+## Step 4: Use the module
+
+Update the greeting remote application to use the federated module.
```

Comment:
I would use the federated module in the host application, rather than the same remote it’s being exposed from

Reply:
hmm okay sure I can update that

Reply:
You might also have to include adding the new `remote` "greeting" to the `host` application's `module-federation.config` file

Reply:
I'll add a comment but this should've been done for you when you create the the host.